thinkhazard/__init__.py

```diff
@@ -212,38 +234,47 @@ def redirect_to_default_language(request):
     return redirect_to_default_language
 
 
-def add_localized_route(config, name, pattern, factory=None, pregenerator=None, **kw):
-    """
-    Create path language aware routing paths.
+class LanguagesPredicate():
 
-    Each route will have /{lang}/ prefix added to them.
+    def __init__(self, val, config):
+        self.val = val
 
-    Optionally, if default language is set, we'll create redirect from an URL
-    without language path component to the URL with the language path
-    component.
-    """
-    orig_factory = factory
-
-    def wrapper_factory(request):
-        lang = request.matchdict["lang"]
-        # determine if this is a supported lang and convert it to a locale,
-        # likely defaulting to your default language if the requested one is
-        # not supported by your app
-        if lang not in request.registry.settings["available_languages"].split():
+    def text(self):
+        return 'languages = %s' % (self.val,)
+
+    phash = text
```

Comment:
Where is phash used?

Reply:
I suppose `phash` is shown in some pyramid log or command line tool output.
See https://github.com/Pylons/pyramid/blob/master/src/pyramid/predicates.py